File: Python/3099_Harshad_Number/3099_Harshad_Number.py
class Solution:
    def sumOfTheDigitsOfHarshadNumber(self, x: int) -> int:
        y = str(x)
        sum = 0
        
        for i in y:
            sum += int(i)
        
        if(x%sum == 0):
            return sum
        else:
            return -1

File: Java/3099_Harshad_Number/3099_Harshad_Number.java
class Solution {
    public int sumOfTheDigitsOfHarshadNumber(int x) {
        int sum = 0, tmp = x;

        while(tmp > 0)
        {
            sum += tmp%10;
            tmp /= 10;
        }

        if(x%sum == 0)
            return sum;
        else
            return -1;

    }
}